tpp-app/src/info/Info.js

```diff
@@ -82,11 +89,44 @@ const cardData = [
 ]
 
 export default function Info ({ navigation }) {
+    const [factCycleArray, setFactCycleArray] = useState([]);
+    useEffect(() => {
+        async function retrieveFactCycle() {
+            var factArray = await GETFactCycle()
+            setFactCycleArray(factArray);
+            console.log(`This is setting factCycleArray on the InfoPage: ${factArray}`)
+
+            if (!factArray){
+                await POSTFactCycle().then(async () => {
+                    factArray = await GETFactCycle();
+                    setFactCycleArray(factArray);
+                })
+            }
+        if (getFullCurrentDateString() != factArray[0]) {
+            POSTFactCycle().then(async () => {
+            factArray = await GETFactCycle();
+            setFactCycleArray(factArray)
+        });
+        }    
+        }
+        retrieveFactCycle()
+    }, [])
+
+
+    let factWhole = factsJSON[factCycleArray[1]]
```

Comment:
For me, the useEffect is not fast enough and factCycleArray is not defined at this time yet, so an error pops. I think you need to do this in the useEffect as well. I was reading this and it might be relevant. https://devtrium.com/posts/async-functions-useeffect